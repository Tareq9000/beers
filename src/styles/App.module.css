.spinner_box{
    position: fixed;
    top: 48%;
    left: 48%;
}
#app_bar{
    padding: 20px;
}
.app_bar_title{
    padding: 5px;
    padding-left: 20px;
}