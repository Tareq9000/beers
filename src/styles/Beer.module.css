.image{
    max-width: 30%;
}
.imagePaper{
    text-align: center;
}
.paper_box{
    padding: 10px;
}
.ingredients_name{
    font-size: 16px;
    display: inline-block;
    margin-right: 5px;
    margin-left: 5px;
}
.ingredients_amount{
    font-size: 12px;
    display: inline-block;
    margin-right: 5px;
    margin-left: 5px;
}
#title{
    display: inline-block;
}
#under_title{
    float: right;
    
}
.info_element_box{
    margin-bottom: 20px;
}
.go_back_button{
    color: white;
    text-decoration: none;
}