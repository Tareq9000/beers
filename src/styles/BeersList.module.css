.beer_list_item{
    color: black;
    text-decoration: none;
}
#list_box{
    min-height: 400px;
}
#page_information{
    font-family: Roboto, Helvetica, Arial, sans-serif;
    display: inline-block;
    margin: 5px;
}