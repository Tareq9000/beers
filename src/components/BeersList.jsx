import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { List, ListItem, ListItemText,  Button } from '@material-ui/core/';
import PropTypes from 'prop-types';

import { createBeer, goPreviousPage, goNextPage } from '../reducers/beerReducer.js';
import styles from '../styles/BeersList.module.css';

export const BeersList = ({ beerCount, beerList, createBeer, per_page, current_page, goNextPage, goPreviousPage }) => {

    let beers = []

    for(let i = 0; i < beerCount; i++) {
        beers.unshift(
            <ListItem key={i+1} button>
                <Link to={'beer'+(i+1)} className={styles.beer_list_item}>
                    <ListItemText className={styles.beer_list_item} primary={beerList[i].name} secondary={beerList[i].contributed_by + ' ' + beerList[i].first_brewed} />
                </Link>
            </ListItem>
        )
    }

    let maxPage = Math.ceil(beers.length / per_page)
    maxPage === 0 ? maxPage = 1 : maxPage = maxPage + 0

    const shouldShowBeers = beers.slice(per_page*(current_page-1), per_page*current_page)

    return(
        <div>
            <Button id={styles.createBeer_button} variant='contained' color='primary' onClick={createBeer}>
                create a beer
            </Button>
            <div id={styles.list_box}>
                <List component='nav' aria-label='main mailbox folders'>
                    {shouldShowBeers}
                </List>
            </div>
            {
                current_page === 1 ?
                <Button id={styles.goPreviousPage_button} variant='contained' color='primary' disabled>
                    {'<'}
                </Button> : 
                <Button id={styles.goPreviousPage_button} variant='contained' color='primary' onClick={goPreviousPage}>
                    {'<'}
                </Button>
            }
            <div id={styles.page_information}>
                {current_page+' of '+ maxPage}
            </div>
            {
                beers[0] ? (Math.ceil(beers[0].key / per_page)) <= current_page ?
                <Button id={styles.goNextPage_button} variant='contained' color='primary' disabled>
                    {'>'} 
                </Button> :
                <Button id={styles.goNextPage_button} variant='contained' color='primary' onClick={goNextPage}>
                    {'>'}
                </Button> :
                <Button id={styles.goNextPage_button} variant='contained' color='primary' disabled>
                    {'>'} 
                </Button>
            }

        </div>
    )
}
BeersList.propTypes = {
    beerCount: PropTypes.number,
    beerList: PropTypes.arrayOf(PropTypes.object),
    per_page: PropTypes.number,
    current_page: PropTypes.number,
    createBeer: PropTypes.func,
    goPreviousPage: PropTypes.func,
    goNextPage: PropTypes.func,
}

const mapStateToProps = ( state ) => {
    const { beerCount, beerList, per_page, current_page } = state.beerReducer
  
    return { beerCount, beerList, per_page, current_page }
}
const mapDispatchToProps = ( dispatch ) => {

    return {
        createBeer: () => (
            dispatch(createBeer())
        ),
        goPreviousPage: () => (
            dispatch(goPreviousPage())
        ),
        goNextPage: () => (
            dispatch(goNextPage())
        ),
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(BeersList)