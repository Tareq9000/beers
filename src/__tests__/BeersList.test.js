import { BeersList } from '../components/BeersList.jsx';
import React from 'react';
import { shallow } from 'enzyme';

import styles from '../styles/BeersList.module.css';

const mockProps = {
    beerCount: 6,
    beerList: [
        {
            name: 'beer',
            contributed_by: 'someone',
            first_brewed: '2020'
        }, {
            name: 'beer',
            contributed_by: 'someone',
            first_brewed: '2020'
        }, {
            name: 'beer',
            contributed_by: 'someone',
            first_brewed: '2020'
        }, {
            name: 'beer',
            contributed_by: 'someone',
            first_brewed: '2020'
        }, {
            name: 'beer',
            contributed_by: 'someone',
            first_brewed: '2020'
        }, {
            name: 'beer',
            contributed_by: 'someone',
            first_brewed: '2020'
        }, 
    ],
    per_page: 5,
    current_page: 1,
    createBeer: jest.fn(),
    goPreviousPage: jest.fn(),
    goNextPage: jest.fn(),
}

describe('BeersList test', () => {

    it('should render properly', () => {
        const wrapper = shallow(<BeersList {...mockProps} />)
        expect(wrapper).toMatchSnapshot();
    })

    it('should call the function createBeer() ', () => {
        const wrapper = shallow(<BeersList {...mockProps} />)

        wrapper.find('#'+styles.createBeer_button).simulate('click')

        expect(mockProps.createBeer).toHaveBeenCalled();
    })

    it('should call the function goPreviousPage() ', () => {
        const thisProps = {...mockProps, current_page: 2,}

        const wrapper = shallow(<BeersList {...thisProps} />)

        wrapper.find('#'+styles.goPreviousPage_button).simulate('click')

        expect(thisProps.goPreviousPage).toHaveBeenCalled();
    })

    it('should call the function goNextPage() ', () => {
        const wrapper = shallow(<BeersList {...mockProps} />)

        wrapper.find('#'+styles.goNextPage_button).simulate('click')

        expect(mockProps.goNextPage).toHaveBeenCalled();
    })

})