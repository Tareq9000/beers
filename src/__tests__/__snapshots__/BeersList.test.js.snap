// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeersList test should render properly 1`] = `ShallowWrapper {}`;
